src/vcpkg-test/dependencies.cpp

```diff
@@ -663,54 +663,56 @@ TEST_CASE ("version parse semver", "[versionplan]")
     check_semver_version(version_build_tag, "1.0.0", "", 1, 0, 0, {});
 
-    auto version_prerelease_build_tag = Versions::SemanticVersion::from_string("1.0.0-alpha+build");
+    auto version_prerelease_build_tag = Versions::semver_from_string("1.0.0-alpha+build");
     check_semver_version(version_prerelease_build_tag, "1.0.0", "alpha", 1, 0, 0, {"alpha"});
 
-    auto version_invalid_incomplete = Versions::SemanticVersion::from_string("1.0-alpha");
+    auto version_invalid_incomplete = Versions::semver_from_string("1.0-alpha");
     CHECK(!version_invalid_incomplete.has_value());
 
-    auto version_invalid_leading_zeroes = Versions::SemanticVersion::from_string("1.02.03-alpha+build");
+    auto version_invalid_leading_zeroes = Versions::semver_from_string("1.02.03-alpha+build");
     CHECK(!version_invalid_leading_zeroes.has_value());
 
-    auto version_invalid_leading_zeroes_in_tag = Versions::SemanticVersion::from_string("1.0.0-01");
+    auto version_invalid_leading_zeroes_in_tag = Versions::semver_from_string("1.0.0-01");
     CHECK(!version_invalid_leading_zeroes_in_tag.has_value());
 
-    auto version_invalid_characters = Versions::SemanticVersion::from_string("1.0.0-alpha#2");
+    auto version_invalid_characters = Versions::semver_from_string("1.0.0-alpha#2");
     CHECK(!version_invalid_characters.has_value());
 }
 
 TEST_CASE ("version parse relaxed", "[versionplan]")
 {
-    auto version_basic = Versions::RelaxedVersion::from_string("1.2.3");
-    check_relaxed_version(version_basic, {1, 2, 3});
+    check_relaxed_version(Versions::relaxed_from_string("1.2.3"), {1, 2, 3});
 
-    auto version_short = Versions::RelaxedVersion::from_string("1");
-    check_relaxed_version(version_short, {1});
+    check_relaxed_version(Versions::relaxed_from_string("1"), {1});
 
-    auto version_long =
-        Versions::RelaxedVersion::from_string("1.20.300.4000.50000.6000000.70000000.80000000.18446744073709551610");
-    check_relaxed_version(version_long, {1, 20, 300, 4000, 50000, 6000000, 70000000, 80000000, 18446744073709551610u});
+    check_relaxed_version(
+        Versions::relaxed_from_string("1.20.300.4000.50000.6000000.70000000.80000000.18446744073709551610"),
+        {1, 20, 300, 4000, 50000, 6000000, 70000000, 80000000, 18446744073709551610u});
 
-    auto version_invalid_characters = Versions::RelaxedVersion::from_string("1.a.0");
-    CHECK(!version_invalid_characters.has_value());
+    check_relaxed_version(Versions::relaxed_from_string("1.0.0-alpha"), {1, 0, 0}, {"alpha"});
 
-    auto version_invalid_identifiers_2 = Versions::RelaxedVersion::from_string("1.1a.2");
-    CHECK(!version_invalid_identifiers_2.has_value());
+    // TODO: Currently this test fails, the produced identifiers are { "alpha-0", "1" }
+    // but my understanding is that the output should be as in this test.
+    check_relaxed_version(Versions::relaxed_from_string("1.0.0-alpha-0.1"), {1, 0, 0}, {"alpha", "0", "1"});
```

Comment:
@ras0219-msft @BillyONeal 

I intentionally pushed this failing test so we can discuss what the correct output should be.

Reply:
It seems that the output was correct all along. 
See https://github.com/semver/semver/issues/78

```suggestion
    check_relaxed_version(Versions::relaxed_from_string("1.0.0-alpha-0.1"), {1, 0, 0}, {"alpha-0", "1"});
```